src/main/java/org/candlepin/model/ConsumerCloudData.java

```diff
@@ -232,46 +234,38 @@ public ConsumerCloudData addCloudOfferingIds(String... cloudOfferingIds) {
      * The IDs are stored internally as a comma-separated string.</p>
      *
      * @param cloudOfferingIds
-     *         the cloud offering IDs to be added; must not be {@code null},
-     *         or contain {@code null} elements; if there is empty list the attribute remains unchanged
+     *  the cloud offering IDs to be added; must not be {@code null}; if there is empty list the attribute
+     *  remains unchanged
      *
      * @return a reference to this {@code ConsumerCloudData} instance
      *
      * @throws IllegalArgumentException
-     *         if {@code cloudOfferingIds} is {@code null}, contains {@code null} elements,
-     *         or if the combined cloud offering IDs exceed the maximum allowed length of 255 characters
+     *  if {@code cloudOfferingIds} is {@code null}, or if the combined cloud offering IDs exceed the
+     *  maximum allowed length of 255 characters
      */
     public ConsumerCloudData addCloudOfferingIds(Collection<String> cloudOfferingIds) {
         if (cloudOfferingIds == null) {
-            throw new IllegalArgumentException("cloudOfferingIds is null or empty");
+            throw new IllegalArgumentException("cloudOfferingIds is null");
```

Comment:
All 3 fields (cloudAccountId, cloudInstanceId, cloudOfferingId) are nullable in the DB, which is according to what we want. So we shouldn't be throwing IAE if they're null. The only field that is non-nullable (except the auto-generated id and the Consumer reference) is the `cloudProviderShortName`.

Reply:
Due to a ton of various abstractions before we get here, that still shouldn't happen, and what you're referencing in this line isn't going to be triggered.

I also find it odd that we don't accept null but we accept empty here, but the issue was triggered by null elements in the collection.

Reply:
Right I agree this shouldn't happen as part of this particular bug, but I thought it is a good idea to codify it on the entity itself that nulls are allowed. And maybe null should be set even in case the input is empty (what is the value of persisting an empty value? nothing afaik)

Reply:
My worry here is that this is spidering into redesigning the entity and having to change/update the initial parsing layer, as well as writing a bunch of tests that were overlooked.

Reply:
Fixed, I guess. Tests updated accordingly.